applications/tari_base_node/src/grpc/base_node_grpc_server.rs

```diff
@@ -529,6 +535,54 @@ impl tari_rpc::base_node_server::BaseNode for BaseNodeGrpcServer {
         Ok(Response::new(response))
     }
 
+    async fn transaction_state(
+        &self,
+        request: Request<tari_rpc::TransactionStateRequest>,
+    ) -> Result<Response<tari_rpc::TransactionStateResponse>, Status>
+    {
+        let request = request.into_inner();
+        let txn: Transaction = request
+            .transaction
+            .ok_or_else(|| Status::invalid_argument("Transaction is empty"))?
+            .try_into()
+            .map_err(|e| Status::invalid_argument(format!("Failed to convert arguments. Invalid transaction.{}", e)))?;
+        debug!(
+            target: LOG_TARGET,
+            "Received TransactionState request from client ({} kernels, {} outputs, {} inputs)",
+            txn.body.kernels().len(),
+            txn.body.outputs().len(),
+            txn.body.inputs().len()
+        );
+
+        let mut handler = self.mempool_service.clone();
+        let res = handler
+            .get_transaction_state_by_excess_sig(txn.body.kernels()[0].excess_sig.clone())
+            .await
+            .map_err(|e| {
+                error!(target: LOG_TARGET, "Error submitting query:{}", e);
+                Status::internal(e.to_string())
+            })?;
+        let response = match res {
+            TxStorageResponse::UnconfirmedPool => tari_rpc::TransactionStateResponse {
+                result: tari_rpc::TransactionLocation::Unconfirmed.into(),
+            },
+            TxStorageResponse::ReorgPool => tari_rpc::TransactionStateResponse {
+                result: tari_rpc::TransactionLocation::Reorg.into(),
+            },
+            TxStorageResponse::NotStoredAlreadySpent => tari_rpc::TransactionStateResponse {
+                result: tari_rpc::TransactionLocation::Mined.into(),
```

Comment:
I don't think this mapping is correct

Reply:
Its what the mempool sends back if a transaction is already spent. If its spent, it has to be mined.